features/support/capybara.rb

```diff
@@ -1,117 +1,59 @@
-end
-
-# monkeypatch to fix
-# not opened for reading (IOError)
-# /cucumber-1.3.20/lib/cucumber/formatter/interceptor.rb:33:in `each'
-# /cucumber-1.3.20/lib/cucumber/formatter/interceptor.rb:33:in `collect'
-# /cucumber-1.3.20/lib/cucumber/formatter/interceptor.rb:33:in `method_missing'
-require 'cucumber/formatter/interceptor'
-class Cucumber::Formatter::Interceptor::Pipe
-  def is_a?(klass)
-    super || klass == IO
-  end
-end
-
-Capybara.register_driver :firefox do |app|
-  Capybara::Selenium::Driver.new(app, browser: :firefox)
-end
-
-Capybara.register_driver :headless_firefox do |app|
-  options = Selenium::WebDriver::Firefox::Options.new
-
-  options.add_argument('-headless')
-  options.add_argument('--window-size=1280,2048')
-
-  driver = Capybara::Selenium::Driver.new(app, browser: :firefox, options: options)
-
-  driver
-end
-
-Capybara.register_driver :firefox do |app|
-  options = Selenium::WebDriver::Firefox::Options.new
-
-  options.add_argument('--window-size=1280,2048')
-
-  driver = Capybara::Selenium::Driver.new(app, browser: :firefox, options: options)
-
-  driver
+  config.server = :webrick # default is `:default` (which uses puma)
 end
 
 Capybara.register_driver :chrome do |app|
-  options = Selenium::WebDriver::Chrome::Options.new
-  options.add_argument('--window-size=1280,2048')
+  options = Selenium::WebDriver::Options.chrome
+  options.add_argument(WINDOW_SIZE_ARG)
   options.add_argument('--disable-search-engine-choice-screen')
   Capybara::Selenium::Driver.new(app, browser: :chrome, options: options)
 end
 
 Capybara.register_driver :headless_chrome do |app|
-  options = Selenium::WebDriver::Options.chrome(
-    logging_prefs: { performance: 'ALL', browser: 'ALL' },
-    perf_logging_prefs: { enableNetwork: true }
-  )
-
+  options = Selenium::WebDriver::Options.chrome
+  options.add_argument(WINDOW_SIZE_ARG)
+  options.add_argument('--disable-search-engine-choice-screen')
   options.add_argument('--headless=new')
   options.add_argument('--no-sandbox')
   options.add_argument('--disable-popup-blocking')
-  options.add_argument('--window-size=1280,2048')
   options.add_argument('--host-resolver-rules=MAP * ~NOTFOUND , EXCLUDE *localhost*')
-  options.add_argument('--disable-search-engine-choice-screen')
   options.add_argument('--disable-gpu')
 
+  options.logging_prefs = { performance: 'ALL', browser: 'ALL' }
+  options.add_option(:perf_logging_prefs, enableNetwork: true)
+
   options.add_preference(:browser, set_download_behavior: { behavior: 'allow' })
 
-  timeout = 120 # default 60
-  client = Selenium::WebDriver::Remote::Http::Default.new(open_timeout: timeout, read_timeout: timeout)
+  Capybara::Selenium::Driver.new(app, browser: :chrome, options: options, timeout: 120)
+end
+
+# NOTE: depending on the scenario you will need to add both
+# @firefox/@headless_firefox AND @javascript tags
+Before '@firefox', '@headless_firefox' do
+  Capybara.javascript_driver = :headless_firefox
+end
```

Comment:
Is this not redundant because of https://github.com/teamcapybara/capybara/blob/3.40.0/lib/capybara/cucumber.rb#L22-L27?

Reply:
Not really.
That snippet from the link sets the `current_driver` to the driver with the name from the tag. It will actually also be executed each time `@firefox` or `@headless_firefox` are set.

Our code, however, sets the `javascript_driver` for **both** firefox tags, this is to override the `config.javascript_driver = :headless_chrome` set globally.